src/components/Login/Login.test.js

```diff
@@ -108,4 +108,20 @@ describe('Login Component', () => {
     const error = await screen.findByText('Unknown error.');
     expect(error).toBeTruthy();
   });
+  it('Successfull login', async () => {
+    render(
+      <FirebaseContext.Provider value={new Firebase()}>
+        <Router>
+          <AuthContext.Provider value={null} >
+            <Login />
+          </AuthContext.Provider>
+        </Router>
+      </FirebaseContext.Provider>
+    );
+    userEvent.type(screen.getByPlaceholderText('Email address'), validEmail);
+    userEvent.type(screen.getByPlaceholderText('Password'), validPassword);
+    userEvent.click(screen.getByText('Log In'));
+    const error = screen.findByTestId('main');
```

Comment:
findByTestId returns a promise, which will always be truthy. In this case, it might work that you remove the router from the test case but istead define
```
const history = jest.fn()
render(
  <FirebaseContext.Provider value={new Firebase()}>
     <Login history={history} />
  </FirebaseContext>
)
```
and at the expect you could check something like
`expect(history).toHaveBeenCalledWith(ROUTES.MAIN);`